### PR TITLE
Add period drop before rule

### PR DESCRIPTION
**Pull request by QiuMM**
Add a new drop rule which has been [discussed](https://github.com/apache/incubator-druid/issues/5869#issuecomment-425671460) in #5869.

### CODE REVIEW

**Comment by QiuMM** [comment]
@nishantmonu51 I'll open a PR in druid-console repo to do that.

**Comment by QiuMM** [comment]
Have adjusted the coordinator console and tested it in a test cluster.

File: server/src/main/resources/static/old-console/js/rules-0.0.2.js [hunk 1]
```diff
@@ -151,6 +155,10 @@ function makeDropByPeriod(rule) {
   return "<span class='rule_label'>period</span><input type='text' name='period' " + "value='" + rule.period + "'/>";
 }
 
+function makeDropBeforeByPeriod(rule) {
+  return "<span class='rule_label'>period</span><input type='text' name='period' " + "value='" + rule.period + "'/>";
```
Comment (drcrallen) [inline server/src/main/resources/static/old-console/js/rules-0.0.2.js]: would it make sense to put something more descriptive so it is clear this `period` is different than the one above it?

File: docs/content/operations/rule-configuration.md [hunk 2]
```diff
@@ -129,6 +129,22 @@ Period drop rules are of the form:
...
+{
+  "type" : "dropBeforeByPeriod",
+  "period" : "P1M"
+}
+```
+
+* `type` - this should always be "dropBeforeByPeriod"
+* `period` - A JSON Object representing ISO-8601 Periods
+
+The interval of a segment will be compared against the specified period. The period is from some time in the past to the current time. The rule matches if the interval before the period. If you just want to retain recent data, you can use this rule to drop the old data that before a specified period.
```
Comment (drcrallen) [inline docs/content/operations/rule-configuration.md]: Just so I'm clear, this is an alternative to doing a load by period followed by a drop forever?

Is this intended to be used in cases where the default load rule is generally used, but you want to configure the maximum age on a per-datasource basis?  If so can that be added to the docs?

Comment (drcrallen) [reply docs/content/operations/rule-configuration.md]: I get the situation now (from the issue description). Form this description it is not clear what scenarios it is desired to use this rule, can either a link to the issue or more detail on when this is appropriate be added here?

Comment (jihoonson) [reply docs/content/operations/rule-configuration.md]: > If you just want to retain recent data, you can use this rule to drop the old data that before a specified period.

Hmm @drcrallen do you think the quote is not enough to explain the desired use case of this rule? Would you please elaborate more on what details you want? Maybe it's better to add like `this rule is effectively dropForever + loadByPeriod (includeFuture = true)`?

Comment (QiuMM) [reply docs/content/operations/rule-configuration.md]: Maybe like below
```  
If you just want to retain recent data, you can use this rule to drop the old data that 
before a specified period and add a loadForever rule to follow it. 
Notes, dropBeforeByPeriod + loadForever is equivalent to dropForever + loadByPeriod (includeFuture = true)
```
?

Comment (jihoonson) [reply docs/content/operations/rule-configuration.md]: Sounds good to me.

File: server/src/main/resources/static/old-console/js/rules-0.0.2.js (hunk 1 above)
Comment (jihoonson) [reply server/src/main/resources/static/old-console/js/rules-0.0.2.js]: Ok. Sounds good.

**Comment by jihoonson** [comment]
The latest change looks good to me. @drcrallen do you have more comments?

**Comment by QiuMM** [comment]
@jihoonson @drcrallen if there are no more comments, I think it can be merged. Then I'll update #6414 and finally open a PR in [druid-console](https://github.com/druid-io/druid-console) to support #6414 and #6415.

**Comment by jihoonson** [comment]
@QiuMM merged. Thanks!

**Pull request merged**